src/framework/preferences/js/Panels.js

```diff
@@ -687,19 +687,50 @@ var fluid_2_0_0 = fluid_2_0_0 || {};
                         value: "{fluid.prefs.panel.textSize}.model.textSize"
                     },
                     range: "{fluid.prefs.panel.textSize}.options.range",
-                    sliderOptions: "{fluid.prefs.panel.textSize}.options.sliderOptions"
+                    sliderOptions: "{fluid.prefs.panel.textSize}.options.sliderOptions",
+                    components: {
+                        slider: {
+                            options: {
+                                listeners: {
+                                    "onCreate.setAriaLabelledBy": {
+                                        "this": "{that}.container",
+                                        "method": "attr",
+                                        "args": ["aria-labelledby", "{textfieldSlider}.options.sliderOptions.labelId"]
+                                    }
+                                }
+                            }
+
+                        },
+                        textfield: {
+                            options: {
+                                listeners: {
+                                    "onCreate.setAriaLabelledBy": {
+                                        "this": "{that}.container",
+                                        "method": "attr",
+                                        "args": ["aria-labelledby", "{textfieldSlider}.options.sliderOptions.labelId"]
+                                    }
+                                }
+                            }
+                        }
+                    }
                 }
             }
         },
         protoTree: {
-            label: {messagekey: "textSizeLabel"},
+            label: {
+                messagekey: "textSizeLabel",
+                decorators: {
+                    attrs: {id: "{that}.options.sliderOptions.labelId"}
+                }
+            },
             multiplier: {messagekey: "multiplier"},
             textSizeDescr: {messagekey: "textSizeDescr"}
         },
         sliderOptions: {
             orientation: "horizontal",
             step: 0.1,
-            range: "min"
+            range: "min",
+            labelId: "textSize-label-" + fluid.allocateGuid()
```

Comment:
I'm not sure it makes sense for the labelId to be in the sliderOptions as it applies to both the slider and the textfield.